zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkConnection.java

```diff
@@ -192,4 +230,53 @@ public String getServers() {
   public void addAuthInfo(String scheme, byte[] auth) {
     _zk.addAuthInfo(scheme, auth);
   }
+
+  private void lookupGetChildrenMethod() {
+    if (_getChildrenMethod != null) {
+      // Method is already cached.
+      return;
+    }
+    try {
+      if (GETCHILDREN_PAGINATION_DISABLED) {
+        lookupNonPaginatedGetChildren();
+      } else {
+        // Lookup the paginated getChildren API
+        _getChildrenMethod =
+            ZooKeeper.class.getMethod("getAllChildrenPaginated", String.class, boolean.class);
+      }
+    } catch (NoSuchMethodException e1) {
+      // Pagination API is not supported, fall back to non-paginated API
+      lookupNonPaginatedGetChildren();
+    }
```

Comment:
nit, how about the following logic?

if (! GETCHILDREN_PAGINATION_DISABLED) {
  try {
    _getChildrenMethod =
            ZooKeeper.class.getMethod("getAllChildrenPaginated", String.class, boolean.class);
    return;
  } catch (...) {
    log...
  }
}
lookupNonPaginatedGetChildren();
return;

Reply:
I thought about it. The reason for this is I'd like to include the logging in the end. If we return before logging, there is no logging.

Since you also prefer this, I have wrapped the code block into a private method so logging is after the method.